pymc_marketing/mmm/components/adstock.py

```diff
@@ -74,12 +80,68 @@ def __init__(
 
         super().__init__(priors=priors, prefix=prefix)
 
+    def sample_curve(
+        self,
+        parameters: xr.Dataset,
+        amount: float = 1.0,
+    ) -> xr.DataArray:
+        """Sample the adstock transformation given parameters.
+
+        Parameters
+        ----------
+        parameters : xr.Dataset
+            Dataset with parameter values.
+        amount : float, optional
+            Amount to apply the adstock transformation to, by default 1.0.
+
+        Returns
+        -------
+        xr.DataArray
+            Adstocked version of the amount.
+
+        """
+
+        time_since = np.arange(0, self.l_max)
+        coords = {
+            "time since exposure": time_since,
+        }
+        x = np.zeros(self.l_max)
+        x[0] = amount
+
+        with pm.Model(coords=coords):
+            var_name = "adstock"
+            pm.Deterministic(
+                var_name,
+                self.apply(x),
+                dims="time since exposure",
+            )
+
+            return pm.sample_posterior_predictive(
```

Comment:
shall we allow additional keyword arguments to sample_plot_posterior (for example to set a random seed?)

Reply:
I had that thought. You'd just want access to the random seed? another other kwargs of interest?
On the random seed, the function is should just be evaluated at each parameter. There should be nothing random about it (i.e. there is no noise in the equation)

Reply:
The documentation for refererence: https://www.pymc.io/projects/docs/en/latest/api/generated/pymc.sample_posterior_predictive.html

To me, none of the other sample_posterior_predictive parameters see exposable

Reply:
I was thinking about the random seed, indeed! But you are right, it is not that relevant!